python/ray/train/base_trainer.py

```diff
@@ -599,8 +600,11 @@ def fit(self) -> Result:
                 path=self._restore_path,
                 trainable=trainable,
                 param_space=param_space,
-                resume_unfinished=True,
-                resume_errored=True,
+                resume_config=_ResumeConfig(
```

Comment:
This makes sense to me. Shall we update it in this user guide: https://docs.ray.io/en/master/train/user-guides/fault-tolerance.html

Probably we need to rename the guide, since it's not only for restoring training, but also for continue training now.

Reply:
I am a bit hesitant to push this feature out in the docs, since it's still a bit awkward to use.

For example, you're forced to change "num epochs" through the partial arguments in a new train loop rather than just modifying the train loop config:

```python
# A bit awkward to use
TorchTrainer.restore(path, train_loop_per_worker=partial(train_fn, num_epochs=10))

# This will not actually update `config["num_epochs"]` in the restored run...
TorchTrainer.restore(path, train_loop_per_worker=partial(train_fn), train_loop_config={"num_epochs": 10})
```

Reply:
Ok. Let's keep it as a experimental feature only for C*va.